dcm2niix_gear/dcm2niix/arrange.py

```diff
@@ -118,6 +118,121 @@ def exit_if_archive_empty(archive_obj):
         os.sys.exit(1)
 
 
+def tally_files(dir_loc, print_out=False):
+    """
+    Function to profile directory. Note current exclusion of {'.DS_Store', '._*'}
+    from file_set and file_tree. # TODO move this exclusion?
+
+    Args:
+        dir_loc: directory to profile
+        [print_out]: print output
+    Returns:
+        file_set: set of file leaves
+        file_tree
+        file_name_path_dict
+    """
+
+    file_set = set()
+    file_name_path_dict = {}
+    file_tree = []
+    gen = os.walk(dir_loc, topdown=True)
+    next_gen = next(gen)
+    while next_gen != StopIteration:
```

Comment:
Usually the way I've seen these loops written is something like:
```
while True:
    try:
        next_gen = next(gen)
    except StopIteration:
        break
    ....
```
I think yours would work, except for the case when the directory is empty maybe?

Reply:
Checked, seems to run:

>>>os.mkdir('./empty_dir')
>>>os.listdir('./empty_dir')
[]

>>>tally_files('./empty_dir', True)
>>>file_set: set()
>>>file_tree: []
>>>file_name_path_dict: {}
(set(), [], {})

> Usually the way I've seen these loops written is something like:
> 
> ```
> while True:
>     try:
>         next_gen = next(gen)
>     except StopIteration:
>         break
>     ....
> ```
> 
> I think yours would work, except for the case when the directory is empty maybe?